pkg/applier/applier.go

```diff
@@ -560,7 +568,7 @@ func (a *supervisor) applyInner(ctx context.Context, objs []client.Object) (map[
 			} else {
 				klog.V(1).Info(e.WaitEvent)
 			}
-			a.addError(processWaitEvent(e.WaitEvent, s.WaitEvent, objStatusMap))
+			a.addError(processWaitEvent(e.WaitEvent, s.WaitEvent, objStatusMap, false))
```

Comment:
Passing bools as function args is usually a smell, IMO. It makes it unclear what the bool means when you call them, without reading the func def. Maybe convert processWaitEvent into a struct method?

Reply:
fair point, I wasn't super pleased with this either. I'll refactor

Reply:
refactored